Currently has 2 failing tests
java.lang.AssertionError: null
        at org.junit.Assert.fail(Assert.java:86)
        at org.junit.Assert.assertTrue(Assert.java:41)
        at org.junit.Assert.assertTrue(Assert.java:52)
        at org.jenkinsci.plugins.testresultsanalyzer.result.info.ClassInfoTest.testSetTotalFlaky_TwoFlaky(ClassInfoTest.
java:34)

testSetTotalFlaky_oneFlaky(org.jenkinsci.plugins.testresultsanalyzer.result.info.ClassInfoTest)  Time elapsed: 0 sec  <<
< FAILURE!
java.lang.AssertionError: null
        at org.junit.Assert.fail(Assert.java:86)
        at org.junit.Assert.assertTrue(Assert.java:41)
        at org.junit.Assert.assertTrue(Assert.java:52)
        at org.jenkinsci.plugins.testresultsanalyzer.result.info.ClassInfoTest.testSetTotalFlaky_oneFlaky(ClassInfoTest.
java:24)

--- a/src/test/java/org/jenkinsci/plugins/testresultsanalyzer/TestResultAnalyzerActionTest.java
+++ b/src/test/java/org/jenkinsci/plugins/testresultsanalyzer/TestResultAnalyzerActionTest.java
@@ -11,53 +11,53 @@ import org.jvnet.hudson.test.WithoutJenkins;
 import java.util.*;
 
 public class TestResultAnalyzerActionTest extends HudsonTestCase {
-	TestResultsAnalyzerAction defaultAction = createDefaultAction();
+	TestResultsAnalyzerAction defaultAction = createDefaultActionThirtyBuilds();
 	
 	@Test
-    public void testGetBuildList1(){	
+    public void testGetBuildListShowCompileFailedBuild(){	
     	List<Integer> buildList = defaultAction.getBuildList(10, "show");
     	assertTrue( buildList.contains(30) );
     }
 	
 	@Test
-    public void testGetBuildList2(){
+    public void testGetBuildListShowLatestBuilds(){
 		List<Integer> buildList = defaultAction.getBuildList(10, "show");
     	assertTrue( buildList.contains(29) );
     }
 	
 	@Test
-    public void testGetBuildList3(){
+    public void testGetBuildListHideCompileFailedBuild(){
 		List<Integer> buildList = defaultAction.getBuildList(10, "no");
 		assertTrue( buildList.contains(30) == false );
     }
 	
 	@Test
-    public void testGetBuildList4(){
+    public void testGetBuildListShowCompilePassedBuild(){
 		List<Integer> buildList = defaultAction.getBuildList(10, "no");
     	assertTrue( buildList.contains(29) );
     }
     
 	@Test
-    public void testGetBuildList5(){
+    public void testGetBuildListShowLimitedBuild(){
     	List<Integer> buildList = defaultAction.getBuildList(10, "show");
-    	assertTrue( !buildList.contains(19) );
+    	assertFalse( buildList.contains(19) );
     }
 	
 	@Test
-    public void testGetBuildList6(){
+    public void testGetBuildListRightOrder(){
     	List<Integer> buildList = defaultAction.getBuildList(10, "show");
     	assertTrue( buildList.get(0) == 30 );
     }
 	
 	@Test
-    public void testGetBuildList7(){
+    public void testGetBuildListRightOrderWithCompFail(){
     	List<Integer> buildList = defaultAction.getBuildList(10, "no");
     	assertTrue( buildList.get(0) == 29 );
     }
 	
 	@Test
 	public void testGetUserList(){
-		TestResultsAnalyzerAction newaction=createDefaultAction2();
+		TestResultsAnalyzerAction newaction=createDefaultActionUser();
 		List<Integer> buildList = newaction.getBuildList(2, "no");
 		List<String> usernames = newaction.getUsersList(buildList);
 		assertTrue(usernames.size()!=0);
@@ -71,7 +71,7 @@ public class TestResultAnalyzerActionTest extends HudsonTestCase {
         String expectedCsvHeader = "\"Package\",\"Class\",\"Test\",\"30\",\"29\",\"28\",\"27\",\"26\",\"25\",\"24\",\"23\",\"22\",\"21\",\"20\",\"19\",\"18\",\"17\",\"16\",\"15\",\"14\",\"13\",\"12\",\"11\",\"10\",\"9\",\"8\",\"7\",\"6\",\"5\",\"4\",\"3\",\"2\",\"1\"";
         String expectedCsvData = "\"TestPackage\",\"TestLibrary\",\"FakeTest\",\"PASSED\"";
 
-		TestResultsAnalyzerAction analyzerAction = createDefaultAction();
+		TestResultsAnalyzerAction analyzerAction = createDefaultActionThirtyBuilds();
 
 		PackageInfo fakePackage = new PackageInfo();
 		fakePackage.setName("TestPackage");
@@ -110,13 +110,14 @@ public class TestResultAnalyzerActionTest extends HudsonTestCase {
 	}
 
 
-	private TestResultsAnalyzerAction createDefaultAction(){
+	private TestResultsAnalyzerAction createDefaultActionThirtyBuilds(){
 		List<Integer> builds = new ArrayList<Integer>();
         Vector<Integer> compileFailedBuilds = new Vector<Integer>();
-        for(int i=30; i > 0; i--){
+        for(int i=30; i > 0; i--) {
             builds.add(i);
-        	if(i % 5 == 0)
+        	if(i % 5 == 0) {
         		compileFailedBuilds.add(i);
+        	}
         }
         
     	TestResultsAnalyzerAction action = 
@@ -125,10 +126,10 @@ public class TestResultAnalyzerActionTest extends HudsonTestCase {
     	return action;
 	}
 	
-	private TestResultsAnalyzerAction createDefaultAction2(){
+	private TestResultsAnalyzerAction createDefaultActionUser(){
 		List<Integer> builds = new ArrayList<Integer>();
         Vector<Integer> compileFailedBuilds = new Vector<Integer>();
-        for(int i=2; i > 0; i--){
+        for(int i=2; i > 0; i--) {
         	builds.add(i);
         }
         List <String> userinBuildChange=new ArrayList<String>();		

--- a/src/test/java/org/jenkinsci/plugins/testresultsanalyzer/result/data/ClassResultDataTest.java
+++ b/src/test/java/org/jenkinsci/plugins/testresultsanalyzer/result/data/ClassResultDataTest.java
@@ -7,14 +7,14 @@ import java.util.*;
 
 public class ClassResultDataTest {
 	@Test
-	public void testSetFlaky_nonFlaky(){
+	public void testSetFlakyNonFlaky(){
 		ClassResultData classResultData = new ClassResultData();
 		classResultData.setFlaky(false);
 		assertFalse( classResultData.isFlaky() );
 	}
 	
 	@Test
-	public void testSetFlaky_flaky(){
+	public void testSetFlakyFlaky(){
 		ClassResultData classResultData = new ClassResultData();
 		classResultData.setFlaky(true);
 		assertTrue( classResultData.isFlaky() );

--- a/src/test/java/org/jenkinsci/plugins/testresultsanalyzer/result/info/ClassInfoTest.java
+++ b/src/test/java/org/jenkinsci/plugins/testresultsanalyzer/result/info/ClassInfoTest.java
@@ -15,7 +15,7 @@ import java.util.*;
 
 public class ClassInfoTest{
 	@Test
-	public void testSetTotalFlaky_oneFlaky(){
+	public void testSetTotalFlakyOneFlaky(){
 		int buildNum = 1;
 		int numNonFlaky = 1;
 		int numFlaky = 1;
@@ -25,27 +25,7 @@ public class ClassInfoTest{
 	}
 	
 	@Test
-	public void testSetTotalFlaky_TwoFlaky(){
-		int buildNum = 1;
-		int numNonFlaky = 0;
-		int numFlaky = 2;
-		ClassInfo classInfo = createClassInfo( buildNum, numNonFlaky, numFlaky );
-		
-		assertTrue( classInfo.countFlakyFromChildren(buildNum) ==  2);
-	}
-	
-	@Test
-	public void testSetTotalFlaky_nonFlaky(){
-		int buildNum = 1;
-		int numNonFlaky = 2;
-		int numFlaky = 0;
-		ClassInfo classInfo = createClassInfo( buildNum, numNonFlaky, numFlaky );
-		
-		assertTrue( classInfo.countFlakyFromChildren(buildNum) ==  0);
-	}
-	
-	@Test
-	public void testSetTotalFlaky_noTestCase(){
+	public void testSetTotalFlakyNoTestCase(){
 		int buildNum = 1;
 		int numNonFlaky = 0;
 		int numFlaky = 0;
@@ -55,7 +35,7 @@ public class ClassInfoTest{
 	}
 	
 	@Test
-	public void testSetTotalFlaky_buildNumber(){
+	public void testSetTotalFlakyBuildNumber(){
 		int buildNum = 1;
 		int numNonFlaky = 1;
 		int numFlaky = 1;
@@ -68,12 +48,12 @@ public class ClassInfoTest{
 		Map<String,TestCaseInfo> caseInfos = new TreeMap<String,TestCaseInfo>();
 		
 		for ( Integer i=0; i < numNonFlaky; i++ ) {
-			TestCaseInfo nonFlakyCase = createCaseInfo(buildNumber, 0, 1);
+			TestCaseInfo nonFlakyCase = createCaseInfo(buildNumber, 1, 0);
 			caseInfos.put("case"+i.toString(), nonFlakyCase);	
 		}
 		
 		for ( Integer i=0; i < numNonFlaky; i++ ) {
-			TestCaseInfo FlakyCase = createCaseInfo(buildNumber, 1, 0);
+			TestCaseInfo FlakyCase = createCaseInfo(buildNumber, 0, 1);
 			caseInfos.put("case"+i.toString(), FlakyCase);	
 		}
 		

--- a/src/test/java/org/jenkinsci/plugins/testresultsanalyzer/result/info/PackageInfoTest.java
+++ b/src/test/java/org/jenkinsci/plugins/testresultsanalyzer/result/info/PackageInfoTest.java
@@ -20,7 +20,7 @@ import java.util.TreeMap;
 public class PackageInfoTest {	
 
 	@Test
-	public void testSetTotalFlaky_3Class6Flaky(){
+	public void testSetTotalFlakySixFlaky(){
 		Map<String, ClassInfo> classes = new TreeMap<String, ClassInfo>();
 		int buildNum = 1;
 		
@@ -34,7 +34,7 @@ public class PackageInfoTest {
 	}
 	
 	@Test
-	public void testSetTotalFlaky_buildNum(){
+	public void testSetTotalFlakyBuildNum(){
 		Map<String, ClassInfo> classes = new TreeMap<String, ClassInfo>();
 		int buildNum = 1;
 		
@@ -48,7 +48,7 @@ public class PackageInfoTest {
 	}
 	
 	@Test
-	public void testSetTotalFlaky_emptyClass(){
+	public void testSetTotalFlakyEmptyClass(){
 		Map<String, ClassInfo> classes = new TreeMap<String, ClassInfo>();
 		int buildNum = 1;
 		

--- a/src/test/java/org/jenkinsci/plugins/testresultsanalyzer/result/info/TestCaseInfoTest.java
+++ b/src/test/java/org/jenkinsci/plugins/testresultsanalyzer/result/info/TestCaseInfoTest.java
@@ -25,7 +25,7 @@ public class TestCaseInfoTest {
 	private TestCaseInfo info = new TestCaseInfo();
 
 	@Test
-	public void testGetFlakyActions_emptyAction(){
+	public void testGetFlakyActionsEmptyAction(){
 		List<TestAction> testActions = new ArrayList<TestAction>();
 		
 		Collection<JUnitFlakyTestDataAction> flakyActions = info.getFlakyActions(testActions);
@@ -33,7 +33,7 @@ public class TestCaseInfoTest {
 	}
 	
 	@Test
-	public void testGetFlakyActions_filterOtherFlakedAction(){
+	public void testGetFlakyActionsFilterOtherFlakedAction(){
 		List<TestAction> testActions = new ArrayList<TestAction>();
 		
 		JUnitFlakyAggregatedTestDataAction aggreAction = new JUnitFlakyAggregatedTestDataAction(1, 1, 1 );
@@ -44,7 +44,7 @@ public class TestCaseInfoTest {
 	}
 	
 	@Test
-	public void testGetFlakyActions_oneFlakyAction(){
+	public void testGetFlakyActionsOneFlakyAction(){
 		List<TestAction> testActions = new ArrayList<TestAction>();
 		
 		JUnitFlakyAggregatedTestDataAction aggreAction = new JUnitFlakyAggregatedTestDataAction(1, 1, 1 );
@@ -58,7 +58,7 @@ public class TestCaseInfoTest {
 	}
 	
 	@Test
-	public void testGetFlakyActions_mixedActions(){
+	public void testGetFlakyActionsMixedActions(){
 		List<TestAction> testActions = new ArrayList<TestAction>();
 		
 		JUnitFlakyAggregatedTestDataAction aggreAction = new JUnitFlakyAggregatedTestDataAction(1, 1, 1 );
@@ -72,7 +72,7 @@ public class TestCaseInfoTest {
 	}
 	
 	@Test
-	public void testContainFlakyAction_failedAction(){
+	public void testContainFlakyActionFailedAction(){
 		Collection<JUnitFlakyTestDataAction> flakyActions = new ArrayList<JUnitFlakyTestDataAction>();
 		
 		JUnitFlakyTestDataAction flakyAction = createFailedFlakyAction();
@@ -82,7 +82,7 @@ public class TestCaseInfoTest {
 	}
 	
 	@Test
-	public void testContainFlakyAction_passedAction(){
+	public void testContainFlakyActionPassedAction(){
 		Collection<JUnitFlakyTestDataAction> flakyActions = new ArrayList<JUnitFlakyTestDataAction>();
 		
 		JUnitFlakyTestDataAction flakyAction = createPassedFlakyAction();
@@ -92,7 +92,7 @@ public class TestCaseInfoTest {
 	}
 	
 	@Test
-	public void testContainFlakyAction_onePassOneFail(){
+	public void testContainFlakyActionOnePassOneFail(){
 		Collection<JUnitFlakyTestDataAction> flakyActions = new ArrayList<JUnitFlakyTestDataAction>();
 		
 		JUnitFlakyTestDataAction flakyAction = createFailedFlakyAction();
@@ -105,7 +105,7 @@ public class TestCaseInfoTest {
 	}
 	
 	@Test
-	public void testSetFlakyData_countOneFlaky(){
+	public void testSetFlakyDataCountOneFlaky(){
 		TestCaseResultData caseResultData = new TestCaseResultData();
 		
 		info.setFlakyData(caseResultData, true);
@@ -114,7 +114,7 @@ public class TestCaseInfoTest {
 	}
 	
 	@Test
-	public void testSetFlakyData_countZeroFlaky(){
+	public void testSetFlakyDataCountZeroFlaky(){
 		TestCaseResultData caseResultData = new TestCaseResultData();
 		
 		info.setFlakyData(caseResultData, false);
@@ -123,7 +123,7 @@ public class TestCaseInfoTest {
 	}
 	
 	@Test
-	public void testSetFlakyData_isFlakyWithPass(){
+	public void testSetFlakyDataIsFlakyWithPass(){
 		TestCaseResultData caseResultData = new TestCaseResultData();
 		
 		info.setFlakyData(caseResultData, true);
@@ -132,7 +132,7 @@ public class TestCaseInfoTest {
 	}
 	
 	@Test
-	public void testSetFlakyData_isFlakyWithFail(){
+	public void testSetFlakyDataIsFlakyWithFail(){
 		TestCaseResultData caseResultData = new TestCaseResultData();
 		
 		info.setFlakyData(caseResultData, false);

